evergreen/evergreen.yml

```
@@ -1289,6 +1314,67 @@ axes:
+        variables:
+           FRAMEWORK: "net452"
+      - id: "netstandard15"
+        display_name: .NET Standard 1.5
+        variables:
+           FRAMEWORK: "netstandard15"
+      - id: "netstandard20"
+        display_name: .NET Standard 2.0
+        variables:
+           FRAMEWORK: "netstandard20"
+      - id: "netstandard21"
+        display_name: .NET Standard 2.1
+        variables:
+           FRAMEWORK: "netstandard21"
+
+task_groups:
+  - name: serverless-task-group
+    setup_group_can_fail_task: true
+    setup_group_timeout_secs: 900 # Setup task can take up to 15 minutes
+    setup_group:
+      # Note: "pre" functions do not run for task group tasks. That's why they are duplicated here.
+      - func: fetch-source
+      - func: prepare-resources
+      - func: windows-fix
+      - func: fix-absolute-paths
+      - func: init-test-results
+      - func: make-files-executable
+      - command: shell.exec
+        params:
+          shell: "bash"
+          silent: true
+          script: |
+            ${PREPARE_SHELL}
+            set +o xtrace
+            SERVERLESS_DRIVERS_GROUP=${SERVERLESS_DRIVERS_GROUP} \
+            SERVERLESS_API_PUBLIC_KEY=${SERVERLESS_API_PUBLIC_KEY} \
+            SERVERLESS_API_PRIVATE_KEY=${SERVERLESS_API_PRIVATE_KEY} \
+              bash ${DRIVERS_TOOLS}/.evergreen/serverless/create-instance.sh
+      - command: expansions.update
+        params:
+          file: serverless-expansion.yml
+    teardown_group:
+      - command: shell.exec
+        params:
+          silent: true
+          script: |
+            ${PREPARE_SHELL}
+            set +o xtrace
+            SERVERLESS_DRIVERS_GROUP=${SERVERLESS_DRIVERS_GROUP} \
+            SERVERLESS_API_PUBLIC_KEY=${SERVERLESS_API_PUBLIC_KEY} \
+            SERVERLESS_API_PRIVATE_KEY=${SERVERLESS_API_PRIVATE_KEY} \
+            SERVERLESS_INSTANCE_NAME=${SERVERLESS_INSTANCE_NAME} \
+              bash ${DRIVERS_TOOLS}/.evergreen/serverless/delete-instance.sh
```

it looks like this step never happens: https://github.com/mongodb-labs/drivers-evergreen-tools/blob/master/.evergreen/serverless/delete-instance.sh#L26 at least I don't see it here: https://evergreen.mongodb.com/task_log_raw/dot_net_driver_serverless_tests_ubuntu__os~ubuntu_1804_auth~auth_ssl~ssl_compressor~zlib_framework~netstandard21_serverless_tests_patch_c04e50a455112fd71df86991024ff66bec1fab97_60f7ff932fbabe15604a8a3d_21_07_21_11_06_22/0?type=T

---

is it mentioned somewhere that it should be `silent`ed? It will hide possible errors that we can see here. Also, at least the patch I checked in java didn't hide this logs

---

We always hide all the sensitive output in the logs. I'll check why Java doesn't do that (mind you: their PR is not merged yet)

---

It doesn't matter in Java because our whole Evergreen project is private.  I think .NET's is too.

---

Yep, they all are private. Then there's no serious reason to keep it silenced.
My opinion is that as far as I'm concerned till this point nothing sensitive has being written into the logs and if we want to make evergreen project public some day, no extra work would be required. Also, if something happens, the build can be rerun with the silence flag lifted, so it is only useful in terms of saving time and catching volatile errors in the scripts.
@jyemin what do you think? Should the restriction be lifted for the advantage of seeing the exact error?

---

I think we pass sensitive data a lot for example here `set-temp-fle-aws-creds.sh` and the only option we have in this case is `set +o xtrace # Disable tracing.` for a particular file. But I think the easiest and probably safest value is what we use here: https://github.com/mongodb/mongo-csharp-driver/blob/master/evergreen/evergreen.yml#L283-L290. This way doesn't print this data to the output.

---

missed

---

Removed silent, but left the tracing disabled.